src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -255,17 +255,38 @@ public E create(E entity, boolean flush) {
     /**
      * @return all entities for a particular type.
      */
-    public CandlepinQuery<E> listAll() {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria();
+    public List<E> listAll() {
+        EntityManager em = this.getEntityManager();
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
```

Comment:
:+1: 

Happy to see the JPA variant of this and not the old Hibernate criteria version.